### PR TITLE
Main page

### PR DESCRIPTION
**Pull request by amohan95**

### CODE REVIEW

File: client/main/components/Main.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
+import React, { PropTypes, Component } from 'react';
+import ReactDOM from 'react-dom';
+import Dropzone from 'react-dropzone';
+import styles from '../../../public/css/main.css';
+import Slider from 'react-slider';
+import PageTemplate from '../../shared/components/PageTemplate.js';
+
+export default class ImageControlArea extends Component {
+  constructor(props) {
```
Comment (vakopian) [inline client/main/components/Main.js]: empty constructor, not really needed

File: client/main/components/Main.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+  render = () => {
+    const {file} = this.props;
+    return file == null ? (
+      <div className='imageControlArea'>
+      </div>
+    ) : (
+      <div className='imageControlArea'>
+          <img className='fullImageView' src={file.preview} />
+          <div className='sliderWrapper'>
```
Comment (vakopian) [inline client/main/components/Main.js]: I'm guessing the `Slider` class doesn't support labels itself? At some point we'll may need to look into a better control that supports labels...

File: client/main/components/Main.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+            <div className='label'>med</div>
+            <div className='label labelRight'>high</div>
+          </div> 
+      </div>
+    );
+  };
+}
+
+export default class SelectedImageArea extends Component {
+  constructor(props) {
```
Comment (vakopian) [inline client/main/components/Main.js]: empty constructor, not really needed

File: client/main/components/Main.js [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    super(props);
+  }
+
+  static propTypes = {
+    file: PropTypes.object,
+  };
+
+  defaultProps = {
+    file: null,
```
Comment (vakopian) [inline client/main/components/Main.js]: the `undefined` default for file would be fine, I don't think there is need to explicitly set the default to `null`

File: client/main/components/Main.js [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+};
+
+export default class FileListElement extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {selected: false};
+  }
+
+  static propTypes = {
+    file: PropTypes.object,
```
Comment (vakopian) [inline client/main/components/Main.js]: add the type for `selected`

File: client/main/components/Main.js [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    );
+  };
+};
+
+export default class ImageListArea extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {fileId: 0}
+  }
+
```
Comment (vakopian) [inline client/main/components/Main.js]: add the static `propTypes` for all the properties

File: client/main/components/Main.js [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+            <div className='label labelLeft'>low</div>
+            <div className='label'>med</div>
+            <div className='label labelRight'>high</div>
+          </div> 
+      </div>
+    );
+  };
+}
+
+export default class SelectedImageArea extends Component {
```
Comment (vakopian) [inline client/main/components/Main.js]: if I'm not mistaken only one class could be `default` exported.

File: client/main/components/Main.js [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+
+  render = () => {
+    const {files, onOpenClick, onListElementClick} = this.props;
+
+    return (
+      <div className='imageListArea'>
+        <div className='centered imageListAreaHeader'>
+          <h1 className='centered'>My Images</h1>
+          Drag and drop images anywhere to upload or <br />
+          <button type='button' onClick={onOpenClick}>
```
Comment (vakopian) [inline client/main/components/Main.js]: Please use the `Button` component from react-bootstrap

File: views/main.html [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <title>Main</title>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
+    <link rel="stylesheet" href="../public/css/bootstrap.min.css">
```
Comment (vakopian) [inline views/main.html]: don't include bootstrap directly. the bootstrap css should be imported together with the used components from `react-bootstrap`

File: package.json [hunk 10]
```diff
@@ -36,7 +39,11 @@
     "css-loader": "^0.23.1",
     "express": "^4.13.4",
     "file-loader": "^0.8.5",
+    "font-awesome": "^4.5.0",
```
Comment (vakopian) [inline package.json]: doesn't look like `font-awesome` is being used currently

File: package.json [hunk 11]
```diff
@@ -20,9 +20,12 @@
   },
   "homepage": "https://github.com/cornsuite/imagepop-ui#readme",
   "dependencies": {
+    "flux": "^2.1.1",
```
Comment (vakopian) [inline package.json]: no `flux` yet

File: package.json [hunk 12]
```diff
@@ -20,9 +20,12 @@
...
     "react": "^0.14.6",
     "react-bootstrap": "^0.28.3",
-    "react-dom": "^0.14.6"
+    "react-dom": "^0.14.6",
+    "react-dropzone": "^3.3.2",
+    "react-router": "^2.0.0"
```
Comment (vakopian) [inline package.json]: no `react-router` yet

File: client/main/components/Main.js [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+  onOpenClick = () => {
+    this.refs.dropzone.open();
+  };
+
+  render = () => {
+    return (
+      <Dropzone className='container-fluid' activeClassName='dropzoneArea dropzoneAreaActive'
+                ref='dropzone' onDrop={this.onDrop} disableClick={true}
+                accept={"image/gif,image/jpeg,image/png"}>
+        <div className='row'>
```
Comment (vakopian) [inline client/main/components/Main.js]: could use the `Row` component from react-bootstrap

File: client/main/components/Main.js [hunk 14]
```diff
@@ -0,0 +1,213 @@
...
+          <div className='dropzoneOverlay'>
+            <h1 className='dropzoneOverlayText'>
+              Drop File to Upload!
+            </h1>
+          </div>
+          <div className='col-sm-4'>
```
Comment (vakopian) [inline client/main/components/Main.js]: could use the `Col` component from react-bootstrap

**Comment by vakopian** [comment]
There are 2 things that this page (and the appropriate component) should also provide.
* when the files are dragged (or picked), they should start uploading to the back-end. That means they need to make the (at least) 2 ajax requests per file.
 1. call the `/app/api/fileupload/start` API endpoint to get a new file upload ID.
 1. call the `/app/api/fileupload/upload?id=ID` API endpoint to actually start the upload.
 See #15 about these APIs.
 Ideally each image thumbnail (i.e. the `FileListElement` control) should also have a small progress bar showing the upload progress.
* There needs to be a submit button on the right side (as part of the `ImageControlArea` component) that will do the final submission for processing. Note that the submit action does not actually push the file since it must already be uploaded asynchronously. Instead it just posts the file id (see #15 and #27)

File: client/main/components/Main.js (hunk 5 above)
Comment (amohan95) [inline client/main/components/Main.js]: I'm not exactly sure what you mean for this -- don't you only set the type for props, not state?

File: app.js [hunk 15]
```diff
@@ -1,9 +1,11 @@
 var express = require('express');
 var app = express();
 var path = require('path');
+var fs = require('fs');
```
Comment (vakopian) [inline app.js]: doesn't look like `fs` is used

File: client/main/components/Main.js [hunk 16]
```diff
@@ -0,0 +1,232 @@
...
+import Slider from 'react-slider';
+import PageTemplate from '../../shared/components/PageTemplate.js';
+import { Modal, Grid, Row, Col, Button } from 'react-bootstrap';
+import request from 'superagent-bluebird-promise';
+
+export class ImageControlArea extends Component {
+  handleSlider = (value) => {
+    console.log(value);
+  };
+
```
Comment (vakopian) [inline client/main/components/Main.js]: Please add `propTypes` for the `file` property

File: client/main/components/Main.js (hunk 5 above)
Comment (vakopian) [inline client/main/components/Main.js]: Line 82 (the first line in the `render()`  method shows that there is another property called `selected`.

File: client/main/components/Main.js [hunk 17]
```diff
@@ -0,0 +1,232 @@
...
+};
+
+export class ImageListArea extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {fileId: 0}
+  }
+
+  static propTypes = {
+    fileId: PropTypes.number,
```
Comment (vakopian) [inline client/main/components/Main.js]: It doesn't look like `fileId` is among the properties. However `files`, `onOpenClick`, `onListElementClick` and `selectedFile` are. Please fix the `propTypes`

**Pull request merged**